src/map/options.ts

```diff
@@ -210,6 +210,9 @@ export class MapOptions extends OptionsGroup {
      * all of them if `trace === undefined`.
      */
     public calculateSizes(rawSizes: number[]): number[] {
+        if (!rawSizes.every((x: number) => x > 0)) {
```

Comment:
This check is at the wrong place. It should only check for negative values when `scaleMode` (below) is `'log'`.

Reply:
That makes more sense! Thank you. Is it also okay to take the log of zero, or is there logic later on to handle if `scaled` is `Infinity`?

Reply:
There is no special logic later, so warning on 0 is also good!